SPECS/systemd/systemd.spec

```diff
@@ -1,62 +1,345 @@
-BuildRequires:  perl-XML-Parser
-BuildRequires:  python3-jinja2
-BuildRequires:  python3-pyelftools
-BuildRequires:  tpm2-tss-devel
-BuildRequires:  util-linux-devel
+BuildRequires:  libselinux-devel
+BuildRequires:  audit-libs-devel
+%if %{without bootstrap}
+BuildRequires:  cryptsetup-devel
+%endif
+BuildRequires:  dbus-devel
+BuildRequires:  /sbin/sfdisk
+# /usr/bin/getfacl is needed by test-acl-util
+BuildRequires:  /usr/bin/getfacl
+BuildRequires:  libacl-devel
+BuildRequires:  gobject-introspection-devel
+BuildRequires:  libblkid-devel
+%if %{with xz}
 BuildRequires:  xz-devel
-BuildRequires:  zstd-devel
-Requires:       %{name}-rpm-macros = %{version}-%{release}
-Requires:       glib
-Requires:       kmod
-Requires:       libcap
-Requires:       libgcrypt
-Requires:       lz4
-Requires:       pam
-Requires:       xz
-Requires(post): audit-libs
-Requires(post): pam
-Requires(post): util-linux-libs
-Obsoletes:      systemd-bootstrap
+BuildRequires:  xz
+%endif
+%if %{with lz4}
+BuildRequires:  lz4-devel
+BuildRequires:  lz4
+%endif
+%if %{with bzip2}
+BuildRequires:  bzip2-devel
+%endif
+%if %{with zstd}
+BuildRequires:  libzstd-devel
+%endif
+BuildRequires:  libidn2-devel
+BuildRequires:  libcurl-devel
+BuildRequires:  kmod-devel
+BuildRequires:  elfutils-devel
+BuildRequires:  openssl-devel
+%if %{with gnutls}
+BuildRequires:  gnutls-devel
+%endif
+%if %{undefined azure}
+BuildRequires:  qrencode-devel
+%endif
+BuildRequires:  libmicrohttpd-devel
+BuildRequires:  libxkbcommon-devel
+BuildRequires:  iptables-devel
+%if %{undefined azure}
+BuildRequires:  pkgconfig(libfido2)
+%endif
+BuildRequires:  pkgconfig(tss2-esys)
+BuildRequires:  pkgconfig(tss2-rc)
+BuildRequires:  pkgconfig(tss2-mu)
+BuildRequires:  pkgconfig(libbpf)
+%if %{undefined azure}
+# This is required for udev tracing
+BuildRequires:  systemtap-sdt-devel
+%endif
+BuildRequires:  libxslt
+BuildRequires:  docbook-style-xsl
+BuildRequires:  pkgconfig
+BuildRequires:  gperf
+BuildRequires:  gawk
+BuildRequires:  tree
+BuildRequires:  hostname
+BuildRequires:  python3
+BuildRequires:  python3-devel
+BuildRequires:  python3dist(jinja2)
+BuildRequires:  python3dist(lxml)
+BuildRequires:  python3dist(pefile)
+%if %{undefined azure}
+BuildRequires:  python3dist(pillow)
+BuildRequires:  python3dist(pytest-flakes)
+%endif
+BuildRequires:  python3dist(pytest)
+BuildRequires:  python3dist(zstd)
+%if 0%{?want_bootloader}
+BuildRequires:  python3dist(pyelftools)
+%endif
+# gzip and lzma are provided by the stdlib
+BuildRequires:  firewalld-filesystem
+BuildRequires:  libseccomp-devel
+BuildRequires:  meson >= 0.43
+BuildRequires:  gettext
+# We use RUNNING_ON_VALGRIND in tests, so the headers need to be available
+BuildRequires:  valgrind-devel
+BuildRequires:  pkgconfig(bash-completion)
+BuildRequires:  perl
+BuildRequires:  perl(IPC::SysV)
+
+%ifnarch %ix86
+# bpftool is not built for i368
+BuildRequires:  bpftool
+%global have_bpf 1
+%endif
+
+%if %{undefined azure}
```

Comment:
I think we can remove all sections with `%{undefined azure}`, since they are never going to be used in our specs, create noise, and bloat the spec. Keeping the spec as complex as this one clean will make it easier for future devs to understand it and make updates.

Reply:
i understand that would be your preference, but i disagree with you, and the change would not actually result in any real change to the built package.